packages/core/src/RenderingEngine/StackViewport.ts

```diff
@@ -1597,60 +1603,61 @@ class StackViewport extends Viewport {
 
   private matchImagesForOverlay(
```

Comment:
I would say to break this function up into smaller pieces that can be called/imported to detect if the match is present or not - the function is very hard to follow given how long it is, but since you are just changing a few things, I think leaving it for now, just noting that it probably needs at least a comment so that people don't have to read it to figure out what is going on.